CONTRIBUTING.md

```diff
@@ -109,9 +179,124 @@ must work with threading**
   `ArrayRef<T>`, or code that handles multiple `ScalarType` types), but for the
   most part avoid them if possible.
 
+&nbsp;
+
+## Testing
+
+### Writing Tests
+To help keep code quality high, ExecuTorch uses a combination of unit tests and
+end-to-end (e2e) tests. If you add a new feature or fix a bug, please add tests
+to ensure that the feature/fix works properly and continues to work properly.
+
+Most directories in the repo already contain test files. In many cases, you can
+add a test to an existing file, and the existing CI jobs will run it will run
+automatically. If you do this, please take a look at the CI job logs to ensure
+that it did actually run.
+
+If it's not clear how to add a test for your PR, take a look at the blame for
+the code you're modifying and find an author who has more context. Ask them
+for their help in the PR comments.
+
+TODO: Explain how to run tests locally without needing to push and wait for CI.
+
+### Continuous Integration
+See https://hud.pytorch.org/hud/pytorch/executorch/main for the current state of
+the CI (continuous integration) jobs. If `main` is broken, consider rebasing
+your PR onto the `viable/strict` branch, which points to the most recent
+all-green commit.
+
+&nbsp;
+
+## Updating Documentation
+
+### APIs
+ExecuTorch documents its APIs using inline code comments: doc strings for
+Python, and Doxygen comments for C++. When modifying or adding an API, be sure
+to modify or add documentation to the interfaces that you change. If the API
+doesn't have inline documentation yet, please help improve the code by adding
+documentation and describing the rest of the piece you modified.
+
+Also search for references to the API you modified under `docs/source` to see if
```

Comment:
elaborate that documents gets mirror to this website: https://pytorch.org/executorch/main/index.html

Reply:
Sure, done